packages/react/src/hooks/track-hooks.ts

```diff
@@ -130,3 +136,93 @@ export function useTracks(sources: Array<Track.Source>, options: UseTracksOption
 
   return pairs;
 }
+
+type UseTilesHookReturnType<T> = T extends Track.Source[]
+  ? TrackParticipantPair[]
+  : T extends TrackSourceWithOptions[]
+  ? MaybeTrackParticipantPair[]
+  : never;
+
+type UseTilesOptions = {
+  updateOnlyOn?: RoomEvent[];
+};
+/**
+ * The useTiles hook allows the looping over subscribed tracks.
+ * Unlike `useTracks`, this hook can also return placeholders alongside `TrackParticipantPair`'s, so they can appear as tiles even without a subscribed track.
+ * Further narrowing the loop items is possible by providing a `filter` function or setting the `excludePinnedTrack` property.
+ *
+ * @example
+ * ```ts
+ * const pairs = useTracks({sources: [Track.Source.Camera], excludePinnedTracks: false})
+ * ```
+ */
+export function useTiles<T extends SourcesArray>(
+  sources: T,
+  options: UseTilesOptions = {},
+): UseTilesHookReturnType<T> {
+  const participants = useParticipants();
+  const sources_ = React.useMemo(() => {
+    return sources.map((s) => (isSourceWitOptions(s) ? s.source : s));
+  }, [JSON.stringify(sources)]);
+
+  const pairs = useTracks(sources_, { updateOnlyOn: options.updateOnlyOn });
+
+  const requirePlaceholder = React.useMemo(() => {
+    const placeholderMap = new Map<Participant['identity'], Track.Source[]>();
+    if (isSourcesWithOptions(sources)) {
+      const sourcesThatNeedPlaceholder = sources
+        .filter((sourceWithOption) => sourceWithOption.withPlaceholder)
+        .map((sourceWithOption) => sourceWithOption.source);
+
+      participants.forEach((participant) => {
+        const sourcesOfSubscribedTracks = participant
+          .getTracks()
+          .map((pub) => pub.track?.source)
+          .filter((trackSource): trackSource is Track.Source => trackSource !== undefined);
+        const placeholderNeededForThisParticipant = Array.from(
+          difference(new Set(sourcesThatNeedPlaceholder), new Set(sourcesOfSubscribedTracks)),
+        );
+        // If the participant needs placeholder add it to the placeholder map.
+        if (placeholderNeededForThisParticipant.length > 0) {
+          placeholderMap.set(participant.identity, placeholderNeededForThisParticipant);
+        }
+      });
+    }
+    console.log({ placeholderMap });
+
+    return placeholderMap;
+  }, [sources, participants, sources_]);
+
+  const tiles = React.useMemo<UseTilesHookReturnType<T>>(() => {
+    if (isSourcesWithOptions(sources)) {
+      const pairs_ = Array.from(pairs) as MaybeTrackParticipantPair[];
+      participants.forEach((participant) => {
+        if (requirePlaceholder.has(participant.identity)) {
+          const sourcesToAddPlaceholder = requirePlaceholder.get(participant.identity) ?? [];
+          sourcesToAddPlaceholder.forEach((placeholderSource) => {
+            console.log(
```

Comment:
let's try to use the logger for new code that needs to log something instead of console.log